src/modules/0.0.23/aws_ecs_fargate/rpcs/deploy_service.ts

```diff
@@ -0,0 +1,84 @@
+import { ECS, UpdateServiceCommandInput } from '@aws-sdk/client-ecs';
+
+import { AwsEcsFargateModule } from '..';
+import { AWS } from '../../../../services/aws_macros';
+import { Context, RpcBase, RpcResponseObject } from '../../../interfaces';
+import { Service } from '../entity';
+
+export class DeployServiceRPC extends RpcBase {
+  module: AwsEcsFargateModule;
+  outputTable = {
+    arn: 'varchar',
+    status: 'varchar',
+    message: 'varchar',
+  } as const;
+
+  async updateService(client: ECS, input: UpdateServiceCommandInput) {
+    const result = await client.updateService(input);
+    return result?.service;
+  }
+
+  call = async (
+    _dbId: string,
+    _dbUser: string,
+    ctx: Context,
+    arn: string,
+  ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
+    if (!arn) {
+      return [
+        {
+          arn: '',
+          status: 'ERROR',
+          message: 'Please provide the ARN of the service to redeploy',
+        },
+      ];
+    }
+    // given the service name, read the details
+    const serviceObj: Service =
+      (await this.module.service.db.read(ctx, this.module.service.generateId({ arn: arn ?? '' }))) ??
+      (await this.module.service.cloud.read(ctx, this.module.service.generateId({ arn: arn ?? '' })));
+
+    if (serviceObj) {
+      const client = (await ctx.getAwsClient(serviceObj.region)) as AWS;
+
+      const service = await this.updateService(client.ecsClient, {
```

Comment:
Does `updateService` force a new deployment on the ECS service?

Reply:
this is was older code was doing. However, there is a "forceNewDeployment" flag that i will add now, that will ensure to force it, i think it makes sense